deltachat-ffi/deltachat.h

```diff
@@ -991,6 +992,33 @@ uint32_t        dc_send_text_msg             (dc_context_t* context, uint32_t ch
 uint32_t dc_send_videochat_invitation (dc_context_t* context, uint32_t chat_id);
 
 
+/**
+ * Send a reaction to message.
+ *
+ * Reaction is a string of emojis separated by spaces. Reaction to a
+ * single message can be sent multiple times. The last reaction
+ * received overrides all previously received reactions. It is
+ * possible to remove all reactions by sending an empty string.
+ *
+ * @memberof dc_context_t
+ * @param context The context object.
+ * @param msg_id ID of the message you react to.
+ * @param reaction A string consisting of emojis separated by spaces.
+ * @return The ID of the message sent out or 0 for errors.
+ */
+uint32_t dc_send_reaction (dc_context_t* context, uint32_t msg_id, char *reaction);
+
+
+/**
+ * Get a structure with reactions to the message.
+ *
+ * @memberof dc_context_t
+ * @param msg_id The message ID to get reactions for.
+ * @return A structure with all reactions to the message.
+ */
+dc_reactions_t* dc_get_msg_reactions (dc_context_t *context, int msg_id);
```

Comment:
i would also be fine with a `dc_get_msg_reactions()` returning JSON if that's easier. it would probably save quite some APIs and all UI have meanwhile json parsing implemented anyway (for webxdc info returning JSON as well).

Reply:
probably also handy for UI is a method to get all reactions emoji+count for a given message. so that the UI can display sth as  ` 👍12   ❤️4 ` without iterating over many API, splitting strings etc.

Reply:
sorry, i edited my initial post while you are already answering. to sum up, these things may be quite complex, it is not only iterating but also sorting and combining emojis may be tricky in detail, see above.

Reply:
> Do you think we need to implement modifiers in reaction picker at all?

i think, this is not needed.

however, if we want to allow to select any emoji as a reaction at some point, the selector then used may offer modifiers (as for signal, at least on ios) - so we have to deal with them somehow. ignoring them seems to be a reasonable approach (signal is also a bit weird here - the last modifier received changes the emoji displayed below the message - so a 10x-black-thumb-up may become a 10x-white-thumb-up by just one white thumb up ... details show the exact amount, however)

Reply:
country flags are also based on modifiers, so I think we should support the most common modifiers eventually. Though I'm unsure whether we should support skin color if that presents grouping problems.

Reply:
also on discord it's the order of what was sent first, which allows for writing small messages:
![image](https://user-images.githubusercontent.com/18725968/194847539-29639f25-804a-4f1c-a528-73df130e7eeb.png)
Though we probably can't guarantee that the order stays the same on every message so a fixed order would indeed make more sense for us.

Reply:
> country flags are also based on modifiers 
> atleast they are comprised from 2 sub emojis

to be more exact:  

- country flags are _not_ based on modifiers in question here ([EMOJI MODIFIER FITZPATRICK](https://emojipedia.org/emoji-modifier-sequence/)).  
- country flags use two [REGIONAL INDICATOR SYMBOL LETTER](https://en.wikipedia.org/wiki/Regional_indicator_symbol), basically the two letter codes of the country, these are no "Modifiers". So eg. `🇨 🇺` becomes "Flag of Cuba".